packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,409 @@
+            let strVoc = '';
+            strVoc = strVoc+`locale: ${locale}\n`;
+            strVoc = strVoc+`namespace: ${namespace}\n`;
+            strVoc = strVoc+'declarations:\n';
+            Object.keys(vocabObject).forEach(decl =>{
+                if (vocabObject[decl].term){
+                    strVoc += `  - ${decl}: ${vocabObject[decl].term}\n`;
+                    const otherProps = Object.keys(vocabObject[decl]).filter((str)=>str!=='term' && str!=='propertyVocabs');
+                    otherProps.forEach(key =>{
+                        strVoc += `    ${key}: ${vocabObject[decl][key]}\n`;
+                    });
+                }
+                if (vocabObject[decl].propertyVocabs && Object.keys(vocabObject[decl].propertyVocabs).length>0){
+                    if (!vocabObject[decl].term){
+                        strVoc += `  - ${decl}: ${decl}\n`;
+                    }
+                    strVoc += '    properties:\n';
+                    Object.keys(vocabObject[decl].propertyVocabs).forEach(prop =>{
+                        strVoc += `      - ${prop}: ${vocabObject[decl].propertyVocabs[prop].term}\n`;
+                        const otherProps = Object.keys(vocabObject[decl].propertyVocabs[prop]).filter((str)=>str!=='term');
+                        otherProps.forEach(key =>{
+                            strVoc += `        ${key}: ${vocabObject[decl].propertyVocabs[prop][key]}\n`;
+                        });
+                    });
+                }
+            });
+            vocabData.push(strVoc);
+        }
+        return vocabData;
+    }
+    /**
+     * Constructs Target object for a given model
+     * @param {string} DCS_VERSION - the version string
+     * @param {string} namespace - the current namespace
+     * @param {Object} obj - the ast of the model
+     * @returns {Object} - the target object
+     * @private
+     */
+    static constructTarget(DCS_VERSION, namespace, obj){
+        const target ={
+            '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.CommandTarget`,
+            'namespace':namespace
+        };
+        if (obj.declaration && obj.declaration!==''){
+            target.declaration = obj.declaration;
+        }
+        if (obj.property && obj.property!==''){
+            target.property = obj.property;
+        }
+        if (obj.mapElement && obj.mapElement!==''){
+            target.mapElement = obj.mapElement;
+        }
+        return target;
+    }
+    /**
+    * parses the extracted decorators and generates arrays of decorator command set and vocabularies
+    *
+    * @param {Object} extractionDictionary - extracted decorators and vocabularies
+    * @param {String} DCS_VERSION - dcs_version
+    * @param {String} locale - locale for vocabularies
+    * @returns {Object} - constructed DCS Dict and processed models ast
+    * @private
+    */
+    static parseDecosAndVocabs(extractionDictionary, DCS_VERSION, locale){
```

Comment:
No need to save few characters here.
```suggestion
    static parseDecoratorsAndVocabularies(extractionDictionary, DCS_VERSION, locale){
```

Reply:
also this could be called transform as well, since parse means something different within the concerto library.
```suggestion
    static transformExtractionDictionary(extractionDictionary, DCS_VERSION, locale){
```